src/main/java/org/candlepin/model/PoolQualifier.java

```diff
@@ -0,0 +1,672 @@
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.model;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Class used for defining criteria that reduces a collection of {@link Pool}s to a subset that meets
+ * the criteria.
+ */
+public class PoolQualifier {
+
+    private Set<String> ids = new HashSet<>();
+    private Set<String> contractNumbers = new HashSet<>();
+    private Set<String> orderNumbers = new HashSet<>();
+    private Set<String> productIds = new HashSet<>();
+    private Set<String> productNames = new HashSet<>();
+    private Set<String> matches = new HashSet<>();
+    private Map<String, List<String>> attributes = new HashMap<>();
+    private Set<String> subscriptionIds = new HashSet<>();
+
+    private Date activeOn;
+    private boolean addFuture = false;
+    private boolean isOnlyFuture = false;
+    private Date after;
+
+    private Consumer consumer;
+    private String ownerId;
+
+    /**
+     * @return
+     *  an unmodifiable set of all the {@link Pool} IDs that qualify a pool
+     */
+    public Set<String> getIds() {
+        return Collections.unmodifiableSet(ids);
+    }
+
+    /**
+     * Adds a {@link Pool} ID condition that a pool must fulfill to be considered qualified.
+     *
+     * @param id
+     *  the ID of a pool that qualifies pools
+     *
+     * @return this instance
+     */
+    public PoolQualifier addId(String id) {
+        if (id == null || id.isBlank()) {
+            return this;
+        }
+
+        this.ids.add(id);
+        return this;
+    }
+
+    /**
+     * Adds multiple {@link Pool} IDs to create conditions that a pool must fulfill to be considered
+     * qualified. If a pool's ID equals one of the provided IDs then it will be considered qualified.
+     *
+     * @param ids
+     *  the IDs of pools that qualifies pools
+     *
+     * @return this instance
+     */
+    public PoolQualifier addIds(Collection<String> ids) {
+        if (ids == null || ids.isEmpty()) {
+            return this;
+        }
+
+        ids.stream()
+            .filter(Objects::nonNull)
+            .forEach(this.ids::add);
+
+        return this;
+    }
+
+    /**
+     * Removes a {@link Pool} ID if it exists
+     *
+     * @param id
+     *  the pool ID to remove
+     *
+     * @return this instance
+     */
+    public PoolQualifier removeId(String id) {
+        if (id == null || id.isBlank()) {
```

Comment:
Oh! That was a good catch! 

I updated all the `add` methods to filter blank values

Reply:
The updated code works, but for future implementations or if you choose to go back through and update this, we could have the bulk add offload to the individual add to keep those restrictions in one place. Since it's a silent rejection, that would reduce the amount of code that needs to be kept in sync:

```
ids.stream()
  .forEach(this::addId);
```

Obviously this starts falling apart if we were to ever update the single-arg add to start rejecting invalid entries with an IAE or some such, so both approaches have pros and cons.